src/main/java/edu/kit/ipd/crowdcontrol/objectservice/payment/PaymentDispatcher.java

```diff
@@ -64,9 +61,9 @@ private void dispatchPayment(Experiment exp) {
                 .forEach((platform, paymentJobs) -> {
                     try {
                         platformManager.payExperiment(platform, exp, paymentJobs);
-                    } catch (TaskOperationException | IllegalWorkerSetException e) {
-                        //TODO maybe notify researcher?
-                        log.fatal(String.format("Unable to pay workers for platform %s", platform), e);
+                        //TODO props to the guy which missed the .join          ^^^^^^^^
+                    } catch (PreActionException e) {
+                        log.fatal("Payment of experiment "+exp+" on platform "+platform+" could not take place!", e.getCause());
```

Comment:
No, since the log message already tells you that the action could not take place, so we want to see the cause, not that is could not take place again.